### ISSUE TITLE
Training with LiTS Dataset

### ISSUE THREAD

**Issue by vishnusanjayrs** [MIC-DKFZ/nnUNet#17]
I used the network architecture of Full res 3d-Unet from ur repo 
norm_op_kwargs = {'eps': 1e-5, 'affine': True}
dropout_op_kwargs = {'p': 0.5, 'inplace': True}
net_nonlin = nn.LeakyReLU
net_nonlin_kwargs = {'negative_slope': 1e-2, 'inplace': True}
net = Generic_UNet(input_channels=1, base_num_features=30, num_classes=2, num_pool=4, num_conv_per_stage=2,
                       feat_map_mul_on_downscale=2, conv_op=nn.Conv3d,
                       norm_op=nn.BatchNorm3d, norm_op_kwargs=norm_op_kwargs,
                       dropout_op=nn.Dropout3d, dropout_op_kwargs=dropout_op_kwargs,
                       nonlin=nn.LeakyReLU, nonlin_kwargs=net_nonlin_kwargs, deep_supervision=False)

and created my own dataset by resampling and getting a random crop of 128^3 as input size while training with liver and lesion masks for prediction and used same optimiser and scheduler settings .Even after training 100k iterations with extensive data augmentations and lr reducing to the floor , the training dice accuracy for lesions did not even increase from 0.01 dice , only liver accuracy was increasing through out and stopped at 0.5 dice . i tried combinations of dice loss , cross entropy and both . Am i missing anything here , i know it is hard to tell with out looking into my code . is there any general advice ,tips  on how to balance the loss ??

**Comment by FabianIsensee** [comment]
Hi,
I would not use dropout. Especially not with that much of a dropout probability. That alone should already make things a lot better. nnU-Net also does not use dropout either.
To be honest though I think it will be quite hard for you to reproduce nnU-Net's results this easily. Half of what nnU-Net is is a highly optimized pipeline where every component goes hand in hand with the next and a lot of seemingly unimportant details matter. Picking some parts of it will not guarantee the same results. I recommend you start out with training nnU-Net's 3d_fullres and work your way from there. If you need help adapting things, please let me know.
Best,
Fabian

**Comment by JiangZongKang** [comment]
hello， @
would you open source code  for ''No new Net''? I am very interested in your research.
thnks, Looking forward to your early reply

**Comment by JiangZongKang** [comment]
> Hi,
> I would not use dropout. Especially not with that much of a dropout probability. That alone should already make things a lot better. nnU-Net also does not use dropout either.
> To be honest though I think it will be quite hard for you to reproduce nnU-Net's results this easily. Half of what nnU-Net is is a highly optimized pipeline where every component goes hand in hand with the next and a lot of seemingly unimportant details matter. Picking some parts of it will not guarantee the same results. I recommend you start out with training nnU-Net's 3d_fullres and work your way from there. If you need help adapting things, please let me know.
> Best,
> Fabian
would you open source code for ''No new Net''? I am very interested in your research.
thnks, Looking forward to your early reply

**Comment by JiangZongKang** [comment]
> Hi,
> I would not use dropout. Especially not with that much of a dropout probability. That alone should already make things a lot better. nnU-Net also does not use dropout either.
> To be honest though I think it will be quite hard for you to reproduce nnU-Net's results this easily. Half of what nnU-Net is is a highly optimized pipeline where every component goes hand in hand with the next and a lot of seemingly unimportant details matter. Picking some parts of it will not guarantee the same results. I recommend you start out with training nnU-Net's 3d_fullres and work your way from there. If you need help adapting things, please let me know.
> Best,
> Fabian

would you open source code for ''No new Net''? I am very interested in your research.
thnks, Looking forward to your early reply

**Comment by JunMa11** [comment]
Hi @JiangZongKang ,

> would you open source code for ''No new Net''?

This respo is the source code of `nnUNet`.  What do you mean by `source code`?

**Comment by JiangZongKang** [comment]
> Hi @JiangZongKang ,
> 
> > would you open source code for ''No new Net''?
> 
> This respo is the source code of `nnUNet`. What do you mean by `source code`?

yes, i konw, but FabianIsensee also have a paper for  ''No new Net'',   i want to source code for ''No new Net''. 
thans for your reply.

**Comment by FabianIsensee** [comment]
Hi,
yeah the naming could have been better. Sorry for the confusion. No New New is something else (although it was based on an early version of nnU-Net) and I cannot make the code public as it is quite the mess. There is no time right now to clean it up :-/
Best,
Fabian

**Comment by FabianIsensee** [comment]
any update on the issue @vishnusanjayrs ?

**Comment by vishnusanjayrs** [comment]
Hi Fabian,

Sorry for not getting back to you earlier (Training time with 3d full res unet is like 3 to 4 days for me :P ). I tried the 3d unet without drop out and the dice score for my validation set improved significantly. I was running some other experiments as well with ur frame work , i think it works as expected if everything is followed using ur frame work.

**Comment by FabianIsensee** [comment]
Thank you for getting back to me. Glad to hear it works now. Like I said, the way dropout was set in your case was not ideal. Yes nnU-Net works best if used the way it is intended to - that is because the entire pipeline is optimized :-)
Best,
Fabian
